optaplanner-core/src/main/java/org/optaplanner/core/config/phase/loop/LoopPhaseConfig.java

```diff
@@ -0,0 +1,102 @@
+import org.optaplanner.core.config.phase.custom.CustomPhaseConfig;
+import org.optaplanner.core.config.util.ConfigUtils;
+
+@XmlType(propOrder = {
+        "phaseConfigList",
+})
+public class LoopPhaseConfig extends PhaseConfig<LoopPhaseConfig> {
+
+    public static final String XML_ELEMENT_NAME = "loop";
+
+    // Warning: all fields are null (and not defaulted) because they can be inherited
+    // and also because the input config file should match the output config file
+
+    @XmlElements({
+            @XmlElement(name = ConstructionHeuristicPhaseConfig.XML_ELEMENT_NAME,
+                    type = ConstructionHeuristicPhaseConfig.class),
+            @XmlElement(name = CustomPhaseConfig.XML_ELEMENT_NAME, type = CustomPhaseConfig.class),
+            @XmlElement(name = ExhaustiveSearchPhaseConfig.XML_ELEMENT_NAME, type = ExhaustiveSearchPhaseConfig.class),
+            @XmlElement(name = LocalSearchPhaseConfig.XML_ELEMENT_NAME, type = LocalSearchPhaseConfig.class),
+            @XmlElement(name = NoChangePhaseConfig.XML_ELEMENT_NAME,
+                    type = NoChangePhaseConfig.class),
+            @XmlElement(name = PartitionedSearchPhaseConfig.XML_ELEMENT_NAME, type = PartitionedSearchPhaseConfig.class),
+            @XmlElement(name = LoopPhaseConfig.XML_ELEMENT_NAME, type = LoopPhaseConfig.class),
+    })
+    private List<PhaseConfig> phaseConfigList = null;
+
+    // ************************************************************************
+    // Constructors and simple getters/setters
+    // ************************************************************************
+
+    public List<PhaseConfig> getPhaseConfigList() {
+        return phaseConfigList;
+    }
+
+    public void setPhaseConfigList(List<PhaseConfig> phaseConfigList) {
+        this.phaseConfigList = phaseConfigList;
+    }
+
+    // ************************************************************************
+    // With methods
+    // ************************************************************************
+
+    public LoopPhaseConfig withPhaseConfigList(List<PhaseConfig> phaseConfigList) {
+        this.phaseConfigList = phaseConfigList;
+        return this;
+    }
+
+    @Override
+    public LoopPhaseConfig inherit(LoopPhaseConfig inheritedConfig) {
+        super.inherit(inheritedConfig);
+        phaseConfigList = ConfigUtils.inheritMergeableListProperty(
+                phaseConfigList, inheritedConfig.getPhaseConfigList());
+        return this;
+    }
+
+    @Override
+    public LoopPhaseConfig copyConfig() {
+        return new LoopPhaseConfig().inherit(this);
+    }
+
+    @Override
+    public void visitReferencedClasses(Consumer<Class<?>> classVisitor) {
+        if (getTerminationConfig() != null) {
+            getTerminationConfig().visitReferencedClasses(classVisitor);
+        }
+        if (phaseConfigList != null) {
+            phaseConfigList.forEach(c -> c.visitReferencedClasses(classVisitor));
```

Comment:
Nitpick: "c" is typically char. Or sometimes class. This makes it easier to read what's going on:
```suggestion
            phaseConfigList.forEach(config -> config.visitReferencedClasses(classVisitor));
```